ffi/diplomat/src/collator.rs

```diff
@@ -0,0 +1,234 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatResult;
+    use icu_collator::{Collator, CollatorOptions};
+
+    use crate::{
+        errors::ffi::ICU4XError, locale::ffi::ICU4XLocale, provider::ffi::ICU4XDataProvider,
+    };
+
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu_collator::Collator, Struct)]
+    pub struct ICU4XCollator(pub Collator);
+
+    #[diplomat::enum_convert(core::cmp::Ordering)]
+    #[diplomat::rust_link(core::cmp::Ordering, Enum)]
+    pub enum ICU4XOrdering {
+        Less = 0,
+        Equal = 1,
+        Greater = 2,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::CollatorOptions, Struct)]
+    pub struct ICU4XCollatorOptions {
+        pub strength: ICU4XCollatorStrength,
+        pub alternate_handling: ICU4XCollatorAlternateHandling,
+        pub case_first: ICU4XCollatorCaseFirst,
+        pub max_variable: ICU4XCollatorMaxVariable,
+        pub case_level: ICU4XCollatorCaseLevel,
+        pub numeric: ICU4XCollatorNumeric,
+        pub backward_second_level: ICU4XCollatorBackwardSecondLevel,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::Strength, Enum)]
+    pub enum ICU4XCollatorStrength {
+        Auto = 0,
+        Primary = 1,
+        Secondary = 2,
+        Tertiary = 3,
+        Quaternary = 4,
+        Identical = 5,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::AlternateHandling, Enum)]
+    pub enum ICU4XCollatorAlternateHandling {
+        Auto = 0,
+        NonIgnorable = 1,
+        Shifted = 2,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::CaseFirst, Enum)]
+    pub enum ICU4XCollatorCaseFirst {
+        Auto = 0,
+        Off = 1,
+        LowerFirst = 2,
+        UpperFirst = 3,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::MaxVariable, Enum)]
+    pub enum ICU4XCollatorMaxVariable {
+        Auto = 0,
+        Space = 1,
+        Punctuation = 2,
+        Symbol = 3,
+        Currency = 4,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::CaseLevel, Enum)]
+    pub enum ICU4XCollatorCaseLevel {
+        Auto = 0,
+        Off = 1,
+        On = 2,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::Numeric, Enum)]
+    pub enum ICU4XCollatorNumeric {
+        Auto = 0,
+        Off = 1,
+        On = 2,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::BackwardSecondLevel, Enum)]
+    pub enum ICU4XCollatorBackwardSecondLevel {
+        Auto = 0,
+        Off = 1,
+        On = 2,
+    }
+
+    impl ICU4XCollator {
+        pub fn try_new(
```

Comment:
nit: should rust_link to try_new_unstable

you should also feel free to ignorelist the try_new_from_foo_provider constructors in ffi_coverage/main.rs (alongside the rest of them)